IBX-8858: Update Solr config between 4.5 and 4.6 (#2486)
ibexa/documentation-developer: docs/update_and_migration/from_4.5/update_from_4.5.md

                    values: []
```

## Update Solr configuration

Solr configuration changes with the addition of new features such as spellchecking.

> [Generate the new configuration files using `generate-solr-config.sh`](install_solr.md#continue-solr-configuration).

Reviewer: Solr configuration might be customized by 3rd-party, so we need to describe actual changes. As developer I would expect at least the same level of detail as provided in PR description: https://github.com/ibexa/solr/pull/52.

Author: e93fb0cf19581dbc9c84852aa86eccd078f9ef4d: I now expose the XML of the component and its assignation to /select; I moved the `generate-solr-config.sh` trick into a note box.

                    values: []
```

## Update Solr configuration

Solr configuration changes with the addition spellchecking feature.

> Configure the `spellcheck` component:

Reviewer 2: ```suggestion
Configure the `spellcheck` component in `solrconfig.xml`:
```

```xml
  <searchComponent name="spellcheck" class="solr.SpellCheckComponent">
    <lst name="spellchecker">
      <str name="name">default</str>
      <str name="field">meta_content__text_t</str>
      <str name="classname">solr.DirectSolrSpellChecker</str>
      <str name="distanceMeasure">internal</str>
      <float name="accuracy">0.5</float>
      <int name="maxEdits">2</int>
      <int name="minPrefix">1</int>
      <int name="maxInspections">5</int>
      <int name="minQueryLength">4</int>
      <float name="maxQueryFrequency">0.01</float>
    </lst>
  </searchComponent>
```

> Add this `spellcheck` component to `/select`:

Reviewer 2: ```suggestion
Add this `spellcheck` component to the `/select` request handler: 
```

```xml
  <requestHandler name="/select" class="solr.SearchHandler">
    <arr name="last-components">
      <str>spellcheck</str>
    </arr>
    <!-- […] -->
  </requestHandler>
```

!!! note

    You can [generate new Solr configuration files using `generate-solr-config.sh`](install_solr.md#continue-solr-configuration),
    and compare with your current configuration to merge those `spellcheck` configurations.

> 

Reviewer 2: ```suggestion

Once `solrconfig.xml` has been changed, solr needs to be restarted for the changes to take affect

```

Author: ```suggestion

Restart Solr for `solrconfig.xml` changes to take effect.

```

                    values: []
```

## Update Solr configuration

> Solr configuration changes with the addition spellchecking feature.

Reviewer 3: ```suggestion
Solr configuration changes with the addition of spellchecking feature.
```

Configure the `spellcheck` component in `solrconfig.xml`:

```xml
  <searchComponent name="spellcheck" class="solr.SpellCheckComponent">
    <lst name="spellchecker">
      <str name="name">default</str>
      <str name="field">meta_content__text_t</str>
      <str name="classname">solr.DirectSolrSpellChecker</str>
      <str name="distanceMeasure">internal</str>
      <float name="accuracy">0.5</float>
      <int name="maxEdits">2</int>
      <int name="minPrefix">1</int>
      <int name="maxInspections">5</int>
      <int name="minQueryLength">4</int>
      <float name="maxQueryFrequency">0.01</float>
    </lst>
  </searchComponent>
```

Add this `spellcheck` component to the `/select` request handler: 

```xml
  <requestHandler name="/select" class="solr.SearchHandler">
    <arr name="last-components">
      <str>spellcheck</str>
    </arr>
    <!-- […] -->
  </requestHandler>
```

!!! note

    You can [generate new Solr configuration files using `generate-solr-config.sh`](install_solr.md#continue-solr-configuration),

>     and compare with your current configuration to merge those `spellcheck` configurations.

Reviewer 3: ```suggestion
    and merge `spellcheck` configuration by comparing new files with your existing setup.
```
